tests/returntypes/transforms/test_split_non_commuting_new.py

```diff
@@ -0,0 +1,300 @@
+
+    def test_expval_non_commuting_observables(self):
+        """Test expval with multiple non-commuting operators"""
+        dev = qml.device("default.qubit", wires=6)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(1)
+            qml.Hadamard(0)
+            qml.PauliZ(0)
+            qml.Hadamard(3)
+            qml.Hadamard(5)
+            qml.T(5)
+            return [
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliX(0)),
+                qml.expval(qml.PauliZ(1)),
+                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+                qml.expval(qml.PauliX(3)),
+                qml.expval(qml.PauliY(5)),
+            ]
+
+        assert all(np.isclose(circuit(), np.array([0.0, -1.0, 0.0, 0.0, 1.0, 1 / np.sqrt(2)])))
+
+    def test_shot_vector_support(self):
+        """Test output is correct when using shot vectors"""
+
+        dev = qml.device("default.qubit", wires=6, shots=(10000, (20000, 2), 30000))
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(1)
+            qml.Hadamard(0)
+            qml.PauliZ(0)
+            qml.Hadamard(3)
+            qml.Hadamard(5)
+            qml.T(5)
+            return [
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliX(0)),
+                qml.expval(qml.PauliZ(1)),
+                qml.expval(
+                    qml.PauliY(0) @ qml.PauliY(1) @ qml.PauliZ(3) @ qml.PauliY(4) @ qml.PauliX(5)
+                ),
+                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+                qml.expval(qml.PauliX(3)),
+                qml.expval(qml.PauliY(5)),
+            ]
+
+        assert np.allclose(
+            circuit(), np.array([0.0, -1.0, 0.0, 0.0, 0.0, 1.0, 1 / np.sqrt(2)]), atol=0.05
+        )
+
+
+# Autodiff tests
+exp_res = np.array([0.77015115, -0.47942554, 0.87758256])
+exp_grad = np.array(
+    [[-4.20735492e-01, -4.20735492e-01], [-8.77582562e-01, 0.0], [-4.79425539e-01, 0.0]]
+)
+
+
+class TestAutodiffSplitNonCommuting:
+    """Autodiff tests for all frameworks"""
+
+    @pytest.mark.autograd
+    def test_split_with_autograd(self):
+        """Test that results after splitting are still differentiable with autograd"""
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev, interface="autograd")
+        def circuit(params):
+            qml.RX(params[0], wires=0)
+            qml.RY(params[1], wires=1)
+            return (
+                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+                qml.expval(qml.PauliY(0)),
+                qml.expval(qml.PauliZ(0)),
+            )
+
+        params = pnp.array([0.5, 0.5])
+        res = circuit(params)
+        grad = qml.jacobian(circuit)(params)
+        assert all(np.isclose(res, exp_res))
+        assert all(np.isclose(grad, exp_grad).flatten())
+
+    # TODO: Currently not possible to jit multiple expvals
```

Comment:
This comment is a bit vague: is this not possible with:
* The new return types;
* `split_non_commuting` or
* Something else?

Reply:
Good catch. I think the comment refers to multiple returns with jitting, but I think this is solved with the new return types.

Added a jax-jit test that verifies that it works 👍 .